go/nbs/table_reader.go

```diff
@@ -38,21 +40,28 @@ func newTableReader(buff []byte, r io.ReaderAt, readAmpThresh uint64) tableReade
 	pos -= uint64Size
 
 	pos -= uint32Size
-	tr.chunkCount = binary.BigEndian.Uint32(buff[pos:])
+	chunkCount := binary.BigEndian.Uint32(buff[pos:])
 
 	// index
-	suffixesSize := uint64(tr.chunkCount) * addrSuffixSize
+	suffixesSize := uint64(chunkCount) * addrSuffixSize
 	pos -= suffixesSize
-	tr.suffixes = buff[pos : pos+suffixesSize]
+	suffixes := make([]byte, suffixesSize)
+	copy(suffixes, buff[pos:])
```

Comment:
yeah, i'm hoping we'll get to the point that a table index is just a wrapper around `[]byte` and at that point, it's much easier to have different approaches to managing the lifetime of those slices. but for now, i'd rather just have one approach which is that the index always makes a copy and keeps it in memory.